### PR TITLE
Removed generic exception from phpdoc

### LINKED ISSUE #1304: Remove FilesystemException from annotations
`FilesystemException` should not be annotated in `@throws`.

If you catch e.g. `UnableToWriteFile` from `FilesystemWriter#write()` you cover all errors which can occur in method `write()`. There is no need to annotate the interface as well. You can still catch whole `FilesystemException` as it is inferred from `UnableToWriteFile`.

The problem is that IDE complains now that you should catch also `FilesystemException`:

image

### PR DESCRIPTION
**Pull request by dakur**
Closes #1304

### CODE REVIEW

File: src/FilesystemAdapter.php [hunk 1]
```diff
@@ -7,102 +7,86 @@
 interface FilesystemAdapter
 {
     /**
-     * @throws FilesystemException
+     * @throws UnableToCheckFileExistence
```
Comment (Nyholm) [inline src/FilesystemAdapter.php:10]: I like that you added this line, but dont remove the `FilesystemException`.

**Review by Nyholm — CHANGES_REQUESTED** [review]
I dont think it is a good idea to remove `FilesystemException` from everywhere. The exception can be thrown as stated in https://github.com/thephpleague/flysystem/issues/1304#issuecomment-831944486. 

Also, this brings no clear advantage. 

Im happy with a PR that just adds `UnableToCheckFileExistence` on `fileExists()`

**Comment by dakur** [comment]
> 
> 
> This issue has been automatically marked as stale because it has not had recent activity. It will be closed if no further activity occurs. Thank you for your contributions.

I plan to reply

**Comment by dakur** [comment]
Sorry, I'm not able to finish this. But the idea was sketched in #1304 and https://github.com/thephpleague/flysystem/issues/1304#issuecomment-832474936 even though some of changes in this PR [might not be appropriate](https://github.com/thephpleague/flysystem/pull/1306#discussion_r631588942).

**Pull request closed without merging**